### PR TITLE
T233 ArSync v0.6: Improve tests, sanity checks, helper functions

### PR DESCRIPTION
**Pull request by decentdav**

### CODE REVIEW

File: jest.setup.js [hunk 1]
```diff
@@ -1,5 +1,15 @@
 import { clear } from 'jest-date-mock';
 
+global.console = {
+  ...console,
+  // uncomment to ignore a specific log level
+  // log: jest.fn(),
+  // debug: jest.fn(),
+  // info: jest.fn(),
+  warn: jest.fn(),
```
Comment (decentdav) [inline jest.setup.js:9]: I like this setting myself, but it's probably better practise to comment out line 9 in the repo, right?

File: src/client/arweave/__tests__/create-transaction.test.js [hunk 2]
```diff
@@ -2,10 +2,13 @@ import { advanceTo } from 'jest-date-mock';
 import { postPodcastMetadata } from '../create-transaction';
 // eslint-disable-next-line import/named
 import { addTag, createTransaction } from '../client';
-// TODO: mock unixTimestamp()
-import { unixTimestamp } from '../../../utils';
 import { toTag } from '../utils';
 
+const MOCK_TIMESTAMP = '1234001234';
+jest.mock('../../../utils', () => ({
```
Comment (decentdav) [inline src/client/arweave/__tests__/create-transaction.test.js:8]: Once again I spent far too much time attempting to separate this mock into a `__mocks__` dir. I guess we'll have to live with repeating this code in other tests where `unixTimestamp()` is asserted.

My rationale is that, without mock, tests could fail if `actualTimestamp > expectedTimestamp`.

File: src/client/arweave/__tests__/utils.test.js [hunk 3]
```diff
@@ -129,22 +120,36 @@ describe('mergeEpisodeBatches', () => {
   });
 
   describe('When some batches contain duplicate episodes metadata', () => {
-    const assertResultProps = result => {
```
Comment (decentdav) [inline src/client/arweave/__tests__/utils.test.js:132]: This was a superfluous and inferior workaround.

File: src/client/arweave/create-transaction.js [hunk 4]
```diff
@@ -66,7 +66,8 @@ function episodeTags(newEpisodes, podcast) {
 }
 
 function getMetadataBatchNumber(podcast, firstEpisodeDate, lastEpisodeDate) {
-  if (!(firstEpisodeDate instanceof Date && lastEpisodeDate instanceof Date)) {
+  if (!(firstEpisodeDate instanceof Date && lastEpisodeDate instanceof Date) ||
```
Comment (decentdav) [inline src/client/arweave/create-transaction.js:69]: This only sanity-checks the two boundary `newEpisodes`.
There is already a TODO for early-stage sanity checking of each episode's Date validity, but this late-stage check prevents rare (future) bugs from posting corrupt metadata.

File: src/client/arweave/utils.js [hunk 5]
```diff
@@ -6,15 +6,22 @@ const PLURAL_TAG_MAP = {
   category: 'categories',
   keyword: 'keywords',
 };
-
 const TAG_EXCLUDES = ['Content-Type', 'Unix-Time'];
 
+function sanityCheckedTag() {
```
Comment (decentdav) [inline src/client/arweave/utils.js:11]: As is, this function is called far too many times. We should probably move this elsewhere and only call it once. I'm not sure where/how, though.

File: src/providers/arweave.jsx [hunk 6]
```diff
@@ -14,16 +15,20 @@ function ArweaveProvider({ children }) {
...
+   * @param {(Object|null)} newWallet
+   */
   async function loadNewWallet(newWallet) {
     if (!loadingWallet.current) {
       loadingWallet.current = true;
 
       const userOrDevWallet = newWallet || await arweave.createNewDevWallet();
-      setWallet(userOrDevWallet);
-      setWalletAddress(await arweave.getWalletAddress(userOrDevWallet));
+      if (!valuesEqual(wallet, newWallet)) {
```
Comment (decentdav) [inline src/providers/arweave.jsx:28]: Moot addition for `arlocal` wallets, but good to have in place.

File: src/client/arweave/utils.js (hunk 5 above)
Comment (mmahdigh) [reply src/client/arweave/utils.js:11]: I guess we can have this check somewhere in our CI/CD pipeline down the road?

**Review by mmahdigh — APPROVED** [review]

**Comment by mmahdigh** [comment]
Great job!

File: src/client/arweave/__tests__/create-transaction.test.js (hunk 2 above)
Comment (decentdav) [reply src/client/arweave/__tests__/create-transaction.test.js:8]: I've tried various `spyOn` approaches as well, but the approach used here is the best one I've found.

**Pull request merged**